apps/full-site-editing/full-site-editing-plugin/donations/fetch-status.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+
+const fetchStatus = async () => {
+	try {
+		const result = await apiFetch( {
+			path: '/wpcom/v2/memberships/status',
+			method: 'GET',
+		} );
+		return result;
+	} catch ( error ) {
+		return Promise.reject( error.message );
+	}
+};
```

Comment:
Based on usage, I don't think we need a helper, but an equivalent would be 
```jsx
return await apiFetch( {
		path: '/wpcom/v2/memberships/status',
		method: 'GET',
} ).catch( ( error ) => { return error.message; } )
```

Reply:
True, we don't really need it, I just feel funny putting API calls in the middle of components (and it keeps the `Edit` component as simple and clear as possible). Anyone feel strongly one way or another?

Reply:
> Anyone feel strongly one way or another?

I'm fine with both, but we actually need the memberships status in a few places:

https://github.com/Automattic/wp-calypso/blob/8effb98d9d09d462510b06e02ab47e36d0c8d31d/apps/full-site-editing/full-site-editing-plugin/premium-content/index.js#L48
https://github.com/Automattic/wp-calypso/blob/7c815711378170666c2876d464e2da1d48e5300f/apps/full-site-editing/full-site-editing-plugin/donations/index.js#L22
https://github.com/Automattic/wp-calypso/blob/17c02e4e7f6ab71e76a1945c82266596505adc8e/apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/container/edit.js#L215

A shared helper to be used in all of them might make sense (not a blocker for this PR).

Reply:
Should be fine if we're using this in a few places 👍